### PR TITLE
EG-2375 - batching notary open sourcing.

### PR DESCRIPTION
**Pull request by StefanIliev545**
Open sourced the JPA notary and removed the simple notary implementation. This should yield performance improvements.

### CODE REVIEW

File: node/src/test/kotlin/net/corda/node/services/transactions/UniquenessProviderTests.kt [hunk 1]
```diff
@@ -52,8 +57,9 @@ class UniquenessProviderTests(
         @JvmStatic
         @Parameterized.Parameters(name = "{0}")
         fun data(): Collection<UniquenessProviderFactory> = listOf(
-                PersistentUniquenessProviderFactory(),
-                RaftUniquenessProviderFactory()
+                JPAUniquenessProviderFactory()
+                // TODO: re-enable raft
+                //RaftUniquenessProviderFactory()
```
Comment (StefanIliev545) [inline node/src/test/kotlin/net/corda/node/services/transactions/UniquenessProviderTests.kt]: This one got transferred with the patch. Was it disabled intentionally or just for development?

Comment (thschroeter) [reply node/src/test/kotlin/net/corda/node/services/transactions/UniquenessProviderTests.kt]: It is disabled just for development. Please re-enable the Raft test.

**Comment by thschroeter** [comment]
I've tested the upgrade path from 4.5 to 4.6 with the notary demo sample. I've deployed the sample version 4.5 run notarisations and then upgraded the node and notary to 4.6 and run notarisations without issues.

**Comment by thschroeter** [comment]
@StefanIliev545 let's remove the now unused simple notary implementation as part of this PR please.

**Review by thschroeter — APPROVED** [review]
LGTM

**Review by blsemo — APPROVED** [review]
LGTM - if I understand this correctly, it brings the default single node notary from ENT to OS as well?

**Comment by thschroeter** [comment]
Yes, that's right @blsemo. Thanks for your review.

**Pull request merged**